src/station_cmd.cpp

```diff
@@ -1810,6 +1862,49 @@ static CommandCost FindJoiningRoadStop(StationID existing_stop, StationID statio
 	return FindJoiningBaseStation<Station, STR_ERROR_MUST_REMOVE_ROAD_STOP_FIRST>(existing_stop, station_to_join, adjacent, ta, st);
 }
 
+/**
+ * Calculates cost of new road stops within the area.
+ * @param tile_area Area to check.
+ * @param flags Operation to perform.
+ * @param is_drive_through True if trying to build a drive-through station.
+ * @param is_truck_stop True when building a truck stop, false otherwise.
+ * @param axis Axis of a drive-through road stop.
+ * @param ddir Entrance direction (#DiagDirection) for normal stops. Converted to the axis for drive-through stops.
+ * @param station StationID to be queried and returned if available.
+ * @param rt Road type to build.
+ * @return The cost in case of success, or an error code if it failed.
+ */
+static CommandCost CalculateRoadStopCost(TileArea tile_area, DoCommandFlag flags, bool is_drive_through, bool is_truck_stop, Axis axis, DiagDirection ddir, StationID *est, RoadType rt)
+{
+	CommandCost cost(EXPENSES_CONSTRUCTION);
+
+	bool success = false;
+
+	/* Check every tile in the area. */
+	for (TileIndex cur_tile : tile_area) {
+
+		CommandCost ret = CheckFlatLandRoadStop(TileArea(cur_tile, cur_tile), flags, is_drive_through ? 5 << axis : 1 << ddir, is_drive_through, is_truck_stop, axis, est, rt);
```

Comment:
Are there constants that could replace these magic numbers?

Reply:
I wasn't able to find anything that can.

Reply:
Technically you could do something like the following to get rid of the magic numbers (untested though):
```
uint invalid_dirs = 0;
if (is_drive_through) {
   SetBit(invalid_dirs, AxisToDiagDir(axis));
   SetBit(invalid_dirs, ReverseDiagDir(AxisToDiagDir(axis)));
} else {
   SetBit(invalid_dirs, ddir) {
}
```
Conceptually one might want to introduce `DiagDirBits` as type just as `TrackDirBits`, with the right functions to set bits given an axis or direction. However, I reckon that's out of the scope of this PR. And it probably would only be really useful when there are other places that use such functionality as well.

Reply:
Thank you, your code looks equivalent and it works as far as I could test it. There's a few other spots in the file where bitwise operations are done with plain numbers, like at line 880, but yeah, they are outside the scope of this change.